qiime/core/type/signature.py

```diff
@@ -113,12 +114,20 @@ def decode_parameters(self, **kwargs):
         return params
 
     def check_types(self, **kwargs):
-        for key, (type_, _) in itertools.chain(self.inputs.items(),
-                                               self.parameters.items()):
+        for key, (type_, _) in self.inputs.items():
+            if kwargs[key] is None:
```

Comment:
I think `key` will be in `kwargs` because the user would have either 1) had to supply a parameter, e.g. if it's a required parameter, or 2) they didn't supply the parameter but it has a default value. Note that the previous code assumed that `key` is in `kwargs` so a change probably isn't necessary in this PR.